content/program-testing.md

```diff
@@ -0,0 +1,534 @@
+    &TOKEN_PROGRAM_ID,
+    &mint_keypair.pubkey(),
+    &mint_auth,
+    Some(&mint_auth),
+    9,
+)
+.unwrap();
+```
+
+We use two functions that we've imported from Solana crates to help create the `create_account` and `initialize_mint` instructions.
+
+Next, we need to create/derive the review, comment counter, and user associated token account addresses.
+
+```rust
+// Create review PDA
+let title: String = "Captain America".to_owned();
+const RATING: u8 = 3;
+let review: String = "Liked the movie".to_owned();
+let (review_pda, _bump_seed) =
+    Pubkey::find_program_address(&[payer.pubkey().as_ref(), title.as_bytes()], &program_id);
+
+// Create comment PDA
+let (comment_pda, _bump_seed) =
+    Pubkey::find_program_address(&[review_pda.as_ref(), b"comment"], &program_id);
+
+// Create user associate token account of token mint
+let init_ata_ix: Instruction = create_associated_token_account(
+    &payer.pubkey(),
+    &payer.pubkey(),
+    &mint_keypair.pubkey(),
+);
+
+let user_ata: Pubkey =
+    get_associated_token_address(&payer.pubkey(), &mint_keypair.pubkey());
+```
+
+Once we have all of the accounts initialized, we can put it all together into a single transaction.
+
+```rust
+// Concat data to single buffer
+let mut data_vec = vec![0];
+data_vec.append(
+    &mut (TryInto::<u32>::try_into(title.len()).unwrap().to_le_bytes())
+        .try_into()
+        .unwrap(),
+);
+data_vec.append(&mut title.into_bytes());
+data_vec.push(RATING);
+data_vec.append(
+    &mut (TryInto::<u32>::try_into(review.len())
+        .unwrap()
+        .to_le_bytes())
+    .try_into()
+    .unwrap(),
+);
+data_vec.append(&mut review.into_bytes());
+
+// Create transaction object with instructions, accounts, and input data
+let mut transaction = Transaction::new_with_payer(
+    &[
+      create_mint_acct_ix,
+      init_mint_ix,
+      init_ata_ix,
+      Instruction {
+          program_id: program_id,
+          accounts: vec![
+              AccountMeta::new_readonly(payer.pubkey(), true),
+              AccountMeta::new(review_pda, false),
+              AccountMeta::new(comment_pda, false),
+              AccountMeta::new(mint_keypair.pubkey(), false),
+              AccountMeta::new_readonly(mint_auth, false),
+              AccountMeta::new(user_ata, false),
+              AccountMeta::new_readonly(SYSTEM_PROGRAM_ID, false),
+              AccountMeta::new_readonly(TOKEN_PROGRAM_ID, false),
+          ],
+          data: data_vec,
+      },
+    ],
+    Some(&payer.pubkey()),
+);
+transaction.sign(&[&payer, &mint_keypair], recent_blockhash);
+
+// Process transaction and compare the result
+assert_matches!(banks_client.process_transaction(transaction).await, Ok(_));
+```
+
+You can now run this test with `cargo test-bpf` and if it’s successful, you’ll be able to see the program logs and the final test result in the terminal. It may take a while to compile and run the test. Feel free to take a look at the solution code here.
```

Comment:
@ixmorrow fix: runon
from: You can now run this test with `cargo test-bpf` and if it’s successful,

to: You can now run this test with `cargo test-bpf`. If it’s successful, ...